// orejime@2.2.1/lib/components/app-list.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _appItem = _interopRequireDefault(require("./app-item"));

var AppList = function AppList(_ref) {
  var t = _ref.t,
      ns = _ref.ns,
      apps = _ref.apps,
      consents = _ref.consents,
      onToggle = _ref.onToggle;
  return _react.default.createElement("ul", {
    className: ns('AppList')
  }, apps.map(function (app) {
    var checked = consents[app.name];

    var handleToggle = function handleToggle(value) {
      return onToggle(app, value);
    };

    return _react.default.createElement("li", {
      key: "app-".concat(app.name),
      className: ns("AppList-item AppList-item--".concat(app.name))
    }, _react.default.createElement(_appItem.default, (0, _extends2.default)({
      checked: checked || app.required,
      onToggle: handleToggle,
      t: t,
      ns: ns
    }, app)));
  }));
};

var _default = AppList;
exports.default = _default;
