// orejime@2.2.1/lib/components/categorized-app-list.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _appList = _interopRequireDefault(require("./app-list"));

var pickApps = function pickApps(apps, names) {
  return names.map(function (name) {
    return apps.find(function (app) {
      return app.name === name;
    });
  });
};

var CategorizedAppList = function CategorizedAppList(_ref) {
  var t = _ref.t,
      ns = _ref.ns,
      categories = _ref.categories,
      apps = _ref.apps,
      consents = _ref.consents,
      onToggle = _ref.onToggle;
  return _react.default.createElement("ul", {
    className: ns('CategorizedAppList')
  }, categories.map(function (_ref2) {
    var name = _ref2.name,
        title = _ref2.title,
        description = _ref2.description,
        appNames = _ref2.apps;
    return _react.default.createElement("li", {
      className: ns('CategorizedAppList-item')
    }, _react.default.createElement("h2", {
      className: ns('CategorizedAppList-title')
    }, t(['categories', name, 'title']) || title), _react.default.createElement("p", {
      className: ns('CategorizedAppList-description')
    }, t(['categories', name, 'description']) || description), _react.default.createElement("div", {
      className: ns('CategorizedAppList-apps')
    }, _react.default.createElement(_appList.default, {
      t: t,
      ns: ns,
      apps: pickApps(apps, appNames),
      consents: consents,
      onToggle: onToggle
    })));
  }));
};

var _default = CategorizedAppList;
exports.default = _default;
